utils/fnIt.ts

```diff
@@ -0,0 +1,13 @@
+import { spy } from "ts-mockito";
+
+const getName = <T>(fn: (arg: T) => (...args: any[]) => any ) => {
+    // create an empty instance
+    const dummyInstance = {} as T;
+    // spy it to mark up some properties
+    const dummySpy: T = spy(dummyInstance);
+    // used this 'typed' spy to extract the name
+    return fn(dummySpy).name;
+}
+export function fnIt<T>(fn: (t:T) => (...args:any[]) => any, message : string, test: () => void){
```

Comment:
Exported functions, and public methods/properties on classes, should have a comment to describe what they are and how they're used.
Try not to describe the what the function does internally (mocks etc), and try not to repeat anything that is in the function signature/name. More try to describe what the motivation is for the function, and how it can be used.